gufe/chemicalstate.py

```diff
@@ -60,11 +60,27 @@ def __init__(
         else:
             self._box_vectors = box_vectors
 
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        if self.identifier != other.identifier:
+            return False
+        if not np.array_equal(self.box_vectors, other.box_vectors,
+                              equal_nan=True):  # nan usually compares to false
+            return False
+        if self.components.keys() != other.components.keys():
+            return False
+        for k in self.components:
+            if self.components[k] != other.components[k]:
+                return False
+
+        return True
+
     def __hash__(self):
         return hash(
             (
                 tuple(sorted(self._components.items())),
-                self._box_vectors,
+                self._box_vectors.tobytes(),
                 self._identifier,
             )
```

Comment:
If performance of hashing is a concern, we might want to cache the hash and the use that as the first equality test.